api/src/transactions.rs

```diff
@@ -562,11 +562,11 @@ impl TransactionsApi {
                 BasicError::bad_request_with_code_no_info(err, AptosErrorCode::InvalidInput)
             })?;
         fail_point_poem("endpoint_encode_submission")?;
-        self.context
-            .check_api_output_enabled("Encode submission", &accept_type)?;
         if !self.context.node_config.api.encode_submission_enabled {
             return Err(api_disabled("Encode submission"));
         }
+        self.context
+            .check_api_output_enabled("Encode submission", &accept_type)?;
```

Comment:
Given this is just reordering, no need to do this in PR, but it might be worth also adding a check that the request isn't using `AcceptType::Bcs` in the first place, since if you can handle BCS responses, you shouldn't be using this endpoint anyway.